python/ray/train/v2/_internal/execution/controller/controller.py

```diff
@@ -304,13 +304,30 @@ def _start_worker_group(
 
         # Check for `bundle_label_selector` to influence WorkerGroup scheduling.
         bundle_label_selector = None
+        if isinstance(scaling_config.bundle_label_selector, list):
+            bundle_label_selector = scaling_config.bundle_label_selector[:num_workers]
+        elif isinstance(scaling_config.bundle_label_selector, dict):
+            bundle_label_selector = [
+                scaling_config.bundle_label_selector.copy() for _ in range(num_workers)
+            ]
+        elif scaling_config.bundle_label_selector is not None:
+            raise ValueError(
+                "`bundle_label_selector` must be a list of dictionaries or a single dictionary."
+            )
```

Comment:
Nit: We already validated this config argument, so let's just remove this extra raise